packages/express/src/index.ts

```diff
@@ -0,0 +1,93 @@
+import { IncomingMessage } from 'http';
+
+import NodeClient from '@logto/node';
+import { Request, Response, NextFunction, Router } from 'express';
+
+import ExpressStorage from './storage';
+import { LogtoExpressConfig } from './types';
+
+export type Middleware = (
+  request: Request,
+  response: Response,
+  next: NextFunction
+) => Promise<void>;
+
+export default class LogtoClient {
+  constructor(private readonly config: LogtoExpressConfig) {}
+
+  handleSignIn =
+    (redirectUri = `${this.config.baseUrl}/logto/sign-in-callback`): Middleware =>
+    async (request, response) => {
+      const nodeClient = this.createNodeClient(request, response);
+      await nodeClient.signIn(redirectUri);
+    };
+
+  handleSignInCallback =
+    (redirectTo = this.config.baseUrl): Middleware =>
+    async (request, response) => {
+      const nodeClient = this.createNodeClient(request, response);
+
+      if (request.url) {
+        await nodeClient.handleSignInCallback(`${this.config.baseUrl}${request.originalUrl}`);
+        response.redirect(redirectTo);
+      }
+    };
+
+  handleSignOut =
+    (redirectUri = this.config.baseUrl): Middleware =>
+    async (request, response) => {
+      const nodeClient = this.createNodeClient(request, response);
+      await nodeClient.signOut(redirectUri);
+    };
+
+  handleAuthRoutes = (): Router => {
+    // eslint-disable-next-line new-cap
+    const router = Router();
+
+    router.use('/logto/:action', (request, response, next) => {
+      const { action } = request.params;
+
+      if (action === 'sign-in') {
+        return this.handleSignIn()(request, response, next);
+      }
+
+      if (action === 'sign-in-callback') {
+        return this.handleSignInCallback()(request, response, next);
+      }
+
+      if (action === 'sign-out') {
+        return this.handleSignOut()(request, response, next);
+      }
+
+      response.status(404).end();
+    });
+
+    return router;
+  };
+
+  private createNodeClient(request: IncomingMessage, response: Response) {
+    this.checkSession(request);
```

Comment:
Just curious why don't we check session in Next.js SDK

Reply:
Just like the previous conversation said, our Next.js SDK contains cookie, but express does not.